mev-relay-rs/src/relay.rs

```diff
@@ -411,51 +405,53 @@ impl Relay {
         Ok(())
     }
 
+    // NOTE: we assume that `auction_request` and `signed_submission` have been validated.
     fn insert_bid_if_greater(
         &self,
         auction_request: AuctionRequest,
-        execution_payload: ExecutionPayload,
+        signed_submission: &SignedBidSubmission,
```

Comment:
this is ok, but could also just pass in something like `Option<BlobsBundle>` here

your approach is likely better to maintain over time as things evolve